lib/zookeeper.js

```diff
@@ -247,25 +247,25 @@ Zookeeper.prototype.listBrokers = function (cb) {
 };
 
 
-Zookeeper.prototype.listConsumers = function (groupId) {
-    var that = this;
+Zookeeper.prototype.listConsumers = function (groupId, watch) {
+    var self = this;
     var path = '/consumers/' + groupId + '/ids';
     this.client.getChildren(
         path,
         function () {
-            if (!that.closed) {
-                that.listConsumers(groupId);
+            if (!self.closed && watch) {

```

Comment:
avoid unnecessary nesting watchers, referring to `Zookeeper.prototype.topicExists`